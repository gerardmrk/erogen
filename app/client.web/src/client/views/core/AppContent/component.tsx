import * as React from "react";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import { Switch } from "react-router";
import LoadingDisplay from "./LoadingDisplay";
import routeConfs from "@client/views/conf.routes";
import EnhancedRoute from "@client/views/components/EnhancedRoute";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppContent extends React.PureComponent<Props, State> {
  public render() {
    return (
      <main className={styles.main}>
        <React.Suspense fallback={<LoadingDisplay />}>
          <Switch>
            {routeConfs.map((r, i) => (
              <EnhancedRoute key={i} {...r} />
            ))}
          </Switch>
        </React.Suspense>
      </main>
    );
  }
}

export default AppContent;
