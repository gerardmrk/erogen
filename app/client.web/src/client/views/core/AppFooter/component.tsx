import * as React from "react";
import { Pane } from "evergreen-ui";
import styles from "./styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/Container";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class AppFooter extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Pane className={styles.main} border={"muted"} background={"tint3"}>
        <Container className={styles.container}>{"AppFooter"}</Container>
      </Pane>
    );
  }
}

export default AppFooter;
