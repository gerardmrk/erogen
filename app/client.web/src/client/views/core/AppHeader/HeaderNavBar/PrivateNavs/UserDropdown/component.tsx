import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Icon from "@client/views/components/ui-elements/Icon";
import Image from "@client/views/components/ui-elements/Image";
import Dropdown from "@client/views/components/ui-modules/Dropdown";
import Placeholder from "@client/views/components/ui-elements/Placeholder";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class UserDropdown extends React.PureComponent<Props, State> {
  private onProfileNavClick = () => {
    if (this.props.profile) {
      this.props.history.push(`/user/${this.props.profile.username}`);
    }
  };

  private onDashboardNavClick = () => {
    this.props.history.push("/");
  };

  private onSettingsNavClick = () => {
    this.props.history.push("/settings");
  };

  private onLogoutActionClick = () => {
    this.props.logout();
  };

  public render() {
    const trigger =
      this.props.isLoadingUser || !this.props.profile ? (
        <Icon loading={true} circular={true} bordered={true} name={"spinner"} />
      ) : this.props.profile.displayPicUrl ? (
        <Image
          avatar={true}
          circular={true}
          bordered={true}
          src={this.props.profile.displayPicUrl}
        />
      ) : (
        <Icon circular={true} bordered={true} name={"user outline"} />
      );

    return (
      <Dropdown
        icon={null}
        trigger={trigger}
        direction={"left"}
        pointing={"top"}
      >
        <Dropdown.Menu direction={"left"}>
          <Dropdown.Header
            onClick={this.onProfileNavClick}
            className={styles.dropdownHeader}
          >
            {this.props.isLoadingUser || !this.props.profile ? (
              <Placeholder>
                <Placeholder.Header>
                  <Placeholder.Line length={"full"} />
                </Placeholder.Header>
              </Placeholder>
            ) : (
              <span>{this.props.profile.username}</span>
            )}
          </Dropdown.Header>

          <Dropdown.Divider />

          <Dropdown.Item onClick={this.onProfileNavClick}>
            {this.props.t("navs.private.profile")}
          </Dropdown.Item>

          <Dropdown.Item onClick={this.onDashboardNavClick}>
            {this.props.t("navs.private.dashboard")}
          </Dropdown.Item>

          <Dropdown.Divider />

          <Dropdown.Item onClick={this.onSettingsNavClick}>
            {this.props.t("navs.private.settings")}
          </Dropdown.Item>

          <Dropdown.Item onClick={this.onLogoutActionClick}>
            {this.props.t("navs.private.logout")}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    );
  }
}
