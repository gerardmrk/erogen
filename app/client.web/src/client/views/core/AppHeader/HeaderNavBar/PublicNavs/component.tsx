import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import NavLink from "@client/views/components/NavLink";
import { headerNavsPublic } from "@client/views/conf.navs";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PublicNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <nav className={styles.main}>
        <div className={styles.otherNavsGroup}>
          {headerNavsPublic.map((n, i) => (
            <div className={styles.navBox} key={i}>
              <NavLink path={n.to} label={n.label} />
            </div>
          ))}
        </div>

        <div className={styles.defaultNavsGroup}>
          <div className={styles.navBox}>
            <NavLink
              path={"/login"}
              label={this.props.t("navs.public.login")}
            />
          </div>

          <div className={styles.navBox}>{"/"}</div>

          <div className={styles.navBox}>
            <NavLink
              path={"/register"}
              label={this.props.t("navs.public.register")}
            />
          </div>
        </div>
      </nav>
    );
  }
}
