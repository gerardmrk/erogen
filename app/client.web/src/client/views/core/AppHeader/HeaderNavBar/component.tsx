import * as React from "react";
import loadable from "@loadable/component";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";

const PublicNavs = loadable(() => import("./PublicNavs"));
const PrivateNavs = loadable(() => import("./PrivateNavs"));

export type Props = LocalProps & StoreProps & DispatchProps;
export type State = {};

export class HeaderNavBar extends React.Component<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        {this.props.isAuthenticated ? <PrivateNavs /> : <PublicNavs />}
      </div>
    );
  }
}
