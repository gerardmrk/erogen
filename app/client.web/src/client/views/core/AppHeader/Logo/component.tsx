import * as React from "react";
import styles from "./component.styles.scss";
import logoSVG from "@client/logo.svg";
import { Link } from "react-router-dom";
import { DEFAULT_PUBLIC_PATH } from "@client/views/conf.routes";

export class Logo extends React.PureComponent {
  public render() {
    return (
      <div className={styles.main}>
        <Link to={DEFAULT_PUBLIC_PATH}>
          <img src={logoSVG} alt={"brand logo"} />
        </Link>
      </div>
    );
  }
}

export default Logo;
