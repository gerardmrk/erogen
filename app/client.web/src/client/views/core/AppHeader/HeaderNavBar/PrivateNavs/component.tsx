import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import UserDropdown from "./UserDropdown";
// import Menu from "@client/views/components/ui.collections/Menu";

type Props = LocalProps & StoreProps & DispatchProps;

type State = {};

export class PrivateNavs extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <div className={styles.navs}>{"..."}</div>
        <div className={styles.userdropdown}>
          <UserDropdown />
        </div>
      </div>
    );
  }
}
