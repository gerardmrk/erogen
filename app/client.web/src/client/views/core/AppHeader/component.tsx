import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps, StoreProps, DispatchProps } from ".";
import Container from "@client/views/components/ui.elements/Container";
import Logo from "./Logo";
import HeaderNavBar from "./HeaderNavBar";

export type Props = LocalProps & StoreProps & DispatchProps;

export type State = {};

export class AppHeader extends React.PureComponent<Props, State> {
  public static readonly i18nNamespace = "core_AppHeader";

  public render() {
    return (
      <header className={styles.main}>
        <Container className={styles.container}>
          <div className={styles.logo}>
            <Logo />
          </div>
          <HeaderNavBar />
        </Container>
      </header>
    );
  }
}
