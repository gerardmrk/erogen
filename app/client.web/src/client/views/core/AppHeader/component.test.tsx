import * as React from "react";
import { shallow } from "enzyme";
import { AppHeader, Props } from "./component";

describe("<AppHeader/>", () => {
  describe("isAuthenticated === true", () => {
    let props: Props;
    beforeAll(() => {
      props = { isAuthenticated: true };
    });

    it("renders ok", async () => {
      const wrapper = await shallow(<AppHeader {...props} />);
      expect(wrapper).toExist();
      expect(wrapper).toHaveClassName("main");
    });
  });

  describe("isAuthenticated === false", () => {
    let props: Props;
    beforeAll(() => {
      props = { isAuthenticated: false };
    });

    it("renders ok", async () => {
      const wrapper = await shallow(<AppHeader {...props} />);
      expect(wrapper).toExist();
      expect(wrapper).toHaveClassName("main");
    });
  });
});
