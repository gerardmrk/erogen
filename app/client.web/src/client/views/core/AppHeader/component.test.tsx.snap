// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader/> snapshot 1`] = `
<header
  className="main"
>
  <div
    className="ui container container"
  >
    <div
      className="logo"
    >
      <div
        className="main"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="brand logo"
            src="logo.svg"
          />
        </a>
      </div>
    </div>
    <div
      className="main"
    >
      <nav
        className="main"
      >
        <div
          className="otherNavsGroup"
        >
          <div
            className="navBox"
          >
            <a
              aria-current={null}
              href="/about/product"
              onClick={[Function]}
            >
              <span>
                Product
              </span>
            </a>
          </div>
          <div
            className="navBox"
          >
            <a
              aria-current={null}
              href="/blog"
              onClick={[Function]}
            >
              <span>
                Blog
              </span>
            </a>
          </div>
          <div
            className="navBox"
          >
            <a
              aria-current={null}
              href="/documentation"
              onClick={[Function]}
            >
              <span>
                Docs
              </span>
            </a>
          </div>
          <div
            className="navBox"
          >
            <a
              aria-current={null}
              href="/support"
              onClick={[Function]}
            >
              <span>
                Support
              </span>
            </a>
          </div>
        </div>
        <div
          className="defaultNavsGroup"
        >
          <div
            className="navBox"
          >
            <a
              aria-current={null}
              href="/login"
              onClick={[Function]}
            >
              <span>
                navs.public.login
              </span>
            </a>
          </div>
          <div
            className="navBox"
          >
            /
          </div>
          <div
            className="navBox"
          >
            <a
              aria-current={null}
              href="/register"
              onClick={[Function]}
            >
              <span>
                navs.public.register
              </span>
            </a>
          </div>
        </div>
      </nav>
    </div>
  </div>
</header>
`;
